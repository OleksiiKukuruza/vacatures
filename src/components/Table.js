import PropTypes from 'prop-types';
import styled from 'styled-components';

const Row = styled.div`
  display: flex;
`;

const Cell = styled.div`
  flex: ${props => props.size};
  display: flex;
  justify-content: ${props => props.justifyContent || 'flex-start'};
`;

Cell.propTypes = {
  size: PropTypes.number.isRequired,
  justifyContent: PropTypes.string
};

export { Row, Cell };
