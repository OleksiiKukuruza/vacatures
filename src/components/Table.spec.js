import React from 'react';
import { shallow } from 'enzyme';
import { Cell, Row } from './Table';

const setup = (Component, props = {}) => {
  const wrapper = shallow(<Component {...props} />);

  return {
    wrapper
  };
};

describe('Table', () => {
  it('renders Cell without crashing', () => {
    const { wrapper } = setup(Cell, { size: 10 });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Row without crashing', () => {
    const { wrapper } = setup(Row);
    expect(wrapper).toMatchSnapshot();
  });
});
