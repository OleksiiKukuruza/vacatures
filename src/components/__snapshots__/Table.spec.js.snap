// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders Cell without crashing 1`] = `
.c0 {
  -webkit-flex: 10;
  -ms-flex: 10;
  flex: 10;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

<div
  className="c0"
  size={10}
/>
`;

exports[`Table renders Row without crashing 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="c0"
/>
`;
